src/main/java/cceclipseplugin/core/EclipseRequestManager.java

```diff
@@ -0,0 +1,201 @@
+package cceclipseplugin.core;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.ArrayList;
+import java.util.List;
+
+import org.eclipse.core.resources.IContainer;
+import org.eclipse.core.resources.IFile;
+import org.eclipse.core.resources.IFolder;
+import org.eclipse.core.resources.IProject;
+import org.eclipse.core.resources.IResource;
+import org.eclipse.core.resources.IWorkspaceRoot;
+import org.eclipse.core.resources.ResourcesPlugin;
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.core.runtime.IProgressMonitor;
+import org.eclipse.core.runtime.NullProgressMonitor;
+import org.eclipse.core.runtime.Path;
+import org.eclipse.swt.widgets.Display;
+
+import cceclipseplugin.ui.UIRequestErrorHandler;
+import cceclipseplugin.ui.dialogs.DialogStrings;
+import cceclipseplugin.ui.dialogs.MessageDialog;
+import dataMgmt.DataManager;
+import dataMgmt.models.FileMetadata;
+import dataMgmt.models.ProjectMetadata;
+import requestMgmt.IInvalidResponseHandler;
+import requestMgmt.RequestManager;
+import websocket.IRequestSendErrorHandler;
+import websocket.WSManager;
+import websocket.models.File;
+import websocket.models.Project;
+import websocket.models.Request;
+import websocket.models.requests.FileCreateRequest;
+import websocket.models.requests.FilePullRequest;
+import websocket.models.responses.FileCreateResponse;
+import websocket.models.responses.FilePullResponse;
+
+public class EclipseRequestManager extends RequestManager {
+
+	public EclipseRequestManager(DataManager dataManager, WSManager wsManager,
+			IRequestSendErrorHandler requestSendErrorHandler, IInvalidResponseHandler invalidResponseHandler) {
+		super(dataManager, wsManager, requestSendErrorHandler, invalidResponseHandler);
+	}
+	
+	@Override
+	public void finishSubscribeToProject(long id, File[] files) {
+		IWorkspaceRoot root = ResourcesPlugin.getWorkspace().getRoot();
+		Project p = PluginManager.getInstance().getDataManager().getSessionStorage().getProjectById(id);
+		IProject eclipseProject = root.getProject(p.getName());
+		ProjectMetadata meta =  new ProjectMetadata();
+		meta.setName(p.getName());
+		meta.setProjectID(id);
+		PluginManager.getInstance().getMetadataManager().putProjectMetadata(eclipseProject.getFullPath().toString(), meta);
+		try {
+			if (eclipseProject.exists()) {
+				eclipseProject.delete(true, true, new NullProgressMonitor());
+			}
+			eclipseProject.create(new NullProgressMonitor());
+			eclipseProject.open(new NullProgressMonitor());
```

Comment:
At the moment it doesn't matter because we're not doing anything with the progress monitor. The purpose of the progress monitor is to provide an reference for some sort of corresponding UI element to know about the progress of a certain task, and we don't have anything that would need to use it for now (i.e. a progress bar)